### PR TITLE
[FIX][15.0] mrp: Fix constrains check total cost share of by-products in MO > 100% or < 0% not raise

### PR DESCRIPTION
**Pull request by LeDungViindoo**
Description of the issue/feature this PR addresses:
- Step 1: Create new MO
- Step 2: In page by-products click three dots to show column cost_share in tree by-products
- Step 3: Create new Byproduct and input cost share >100% or <10% or input two  byproducts have cost share is 60% and 70%

Current behavior before PR:

https://user-images.githubusercontent.com/117807675/232964088-ea25087b-5ca0-448f-90c7-2deb00ab49a1.mp4

Desired behavior after PR is merged:

https://user-images.githubusercontent.com/117807675/232964236-8f7f1e67-f004-4c4f-9630-02cd202bbdf6.mp4

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/tests/test_byproduct.py [hunk 1]
```diff
@@ -205,3 +206,59 @@ def test_byproduct_putaway(self):
         finished_move_line = mo.move_finished_ids.filtered(lambda m: m.product_id == self.product_a).move_line_ids
         self.assertEqual(byproduct_move_line.location_dest_id, shelf2_location)
         self.assertEqual(finished_move_line.location_dest_id, shelf2_location)
+
+    def test_check_byproducts(self):
+        """
+        Test the byproducts have total cost_share > 100% or < 0% will raise constrains.
+        """
```
Comment (ticodoo) [inline addons/mrp/tests/test_byproduct.py:213]: Your test is good (thank you for writing one :bow:). Just a little English cleanup

```suggestion
    def test_check_byproducts_cost_share(self):
        """
        Test that byproducts with total cost_share > 100% or a cost_share < 0% 
        will throw a ValidationError
        """
 ```

**Review by ticodoo — COMMENTED** [review]
Hello@LeDungViindoo nice catch/fix! It looks like you need to sign the CLA before we can merge your PR though. Also can you improve your commit message a bit since it's a tricky bug?

```
[FIX] mrp: enforce constrains check for cost share of byproducts in MO

Steps to reproduce:

Step 1: Create new MO
Step 2: In page by-products click three dots to show column cost_share in tree by-products
Step 3: Create new Byproduct and input cost share >100% or <0% or input two byproducts
        where their cost share adds up to more than 100%

Expected result:
Validation error: total byproduct cost_share cannot exceed 100
or Validation error: cost_share values must be positive

Actual result:
Values saves without issue

Issue is due to `move_byproduct_ids` being removed in the mrp_production overrides of
`write()` or `create()` and having the `_compute_move_byproduct_ids` populate its values.
This removal is causing the constrains to not be enforced, therefore we set the constrains
field to `move_finished_ids` since changing of this value will ensure that the values are
correctly checked. 
Note: There will be a decrease in performance since this means the constrains will be
called whenever the MO's product to produce is changed, but hopefully it will be
minimal since there is no simple fix for this issue.
```

**Comment by LeDungViindoo** [comment]
> Hello@LeDungViindoo nice catch/fix! It looks like you need to sign the CLA before we can merge your PR though. Also can you improve your commit message a bit since it's a tricky bug?
> 
> ```
> [FIX] mrp: enforce constrains check for cost share of byproducts in MO
> 
> Steps to reproduce:
> 
> Step 1: Create new MO
> Step 2: In page by-products click three dots to show column cost_share in tree by-products
> Step 3: Create new Byproduct and input cost share >100% or <0% or input two byproducts
>         where their cost share adds up to more than 100%
> 
> Expected result:
> Validation error: total byproduct cost_share cannot exceed 100
> or Validation error: cost_share values must be positive
> 
> Actual result:
> Values saves without issue
> 
> Issue is due to `move_byproduct_ids` being removed in the mrp_production overrides of
> `write()` or `create()` and having the `_compute_move_byproduct_ids` populate its values.
> This removal is causing the constrains to not be enforced, therefore we set the constrains
> field to `move_finished_ids` since changing of this value will ensure that the values are
> correctly checked. 
> Note: There will be a decrease in performance since this means the constrains will be
> called whenever the MO's product to produce is changed, but hopefully it will be
> minimal since there is no simple fix for this issue.
> ```

I corrected it according to what you said

**Comment by LeDungViindoo** [comment]
> Hello @LeDungViindoo it looks like you need to squash your 2 commits into 1 and add your CLA commit still

Hi, I squashed 2 commits into 1 and my Viindoo company is creating a PR to update the CLA file. Please wait.

**Comment by royleviindoo** [comment]
> > Hello @LeDungViindoo it looks like you need to squash your 2 commits into 1 and add your CLA commit still
> 
> Hi @ticodoo, I squashed 2 commits into 1 and my Viindoo company created a PR to update the CLA file.
> 
> PR update CLA: #119586

@LeDungViindoo  rebase your code, please

**Comment by LeDungViindoo** [comment]
> Hello @LeDungViindoo it looks like you need to squash your 2 commits into 1 and add your CLA commit still

Can you check my PR again? Please

**Comment by ticodoo** [comment]
thanks!

@robodoo r+

**Pull request closed without merging**